src/resolve/resolved-dns-rr.c

```diff
@@ -2129,6 +2129,42 @@ static int txt_to_json(DnsTxtItem *items, JsonVariant **ret) {
         return r;
 }
 
+static int svc_params_to_json(DnsSvcParam *params, JsonVariant **ret) {
+        JsonVariant **elements = NULL;
+        size_t n = 0;
+        int r;
+
+        assert(ret);
+
+        LIST_FOREACH(params, i, params) {
+                _cleanup_free_ char *param = NULL;
+                if (!GREEDY_REALLOC(elements, n + 1)) {
+                        r = -ENOMEM;
+                        goto finalize;
+                }
+
+                param = format_svc_param(i);
```

Comment:
this is a bit chickening out. we probably should also propagate this as json

Reply:
That's true... the trouble is there will be keys we don't recognize and so we won't know the structure of the param values, so I'm not sure it makes sense to split e.g. alpn list into a json list. We could have key = value pairs though for each svcparam. I guess I'll do that.

Reply:
i'd be fine with chickening out here if we just leave the data in binary, i.e. expose this as base64?